frontstage/views/surveys.py

```diff
@@ -1,37 +1,164 @@
 from frontstage.common.authorisation import jwt_authorization
+from frontstage.common.cryptographer import Cryptographer
 from frontstage.exceptions.exceptions import ApiError
-
+from frontstage.models import EnrolmentCodeForm
 
 logger = wrap_logger(logging.getLogger(__name__))
 surveys_bp = Blueprint('surveys_bp', __name__,
                        static_folder='static', template_folder='templates/surveys')
+cryptographer = Cryptographer()
 
 
 @surveys_bp.route('/', methods=['GET'])
 @jwt_authorization(request)
 def logged_in(session):
     party_id = session.get('party_id')
     surveys_list = get_surveys_list(party_id, 'todo')
-    return render_template('surveys/surveys-todo.html', _theme='default', surveys_list=surveys_list)
+    template_data = {"survey": {"justAdded": "false"}}
+
+    sorted_surveys_list = sorted(surveys_list, key=lambda k: k['collection_exercise']['scheduledReturnDateTime'],
+                                 reverse=True)
+
+    # Checks if new survey is added and renders "survey added" notification
+    if request.args.get('new_survey') and ['NEW_SURVEY_NOTIF_HIGHLIGHTING'] == 1:
+        logger.info('New survey added highlighted')
+        template_data = {"survey": {"justAdded": "true"}}
+        return render_template('surveys/surveys-todo.html', _theme='default',
+                               data=template_data, sorted_surveys_list=sorted_surveys_list,
+                               just_added_case_id=request.args.get('case_id'))
+
+    return render_template('surveys/surveys-todo.html', data=template_data,
+                           _theme='default', sorted_surveys_list=sorted_surveys_list)
 
 
 @surveys_bp.route('/history', methods=['GET'])
 @jwt_authorization(request)
 def surveys_history(session):
     party_id = session['party_id']
     surveys_list = get_surveys_list(party_id, 'history')
-    return render_template('surveys/surveys-history.html',  _theme='default',
+    template_data = {"survey": {"justAdded": "false"}}
+    return render_template('surveys/surveys-history.html',  _theme='default', data=template_data,
                            surveys_list=surveys_list, history=True)
 
 
+@surveys_bp.route('/add-survey', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def add_survey(session):
+    form = EnrolmentCodeForm(request.form)
+
+    if request.method == 'POST' and form.validate():
+        logger.info('Enrolment code submitted')
+        enrolment_code = request.form.get('enrolment_code').lower()
+        request_data = {
+            'enrolment_code': enrolment_code,
+            'initial': True
+        }
+        response = api_call('POST', app.config['VALIDATE_ENROLMENT'], json=request_data)
+
+        # Handle API errors
+        if response.status_code == 404:
+            logger.info('Enrolment code not found')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
```

Comment:
These return statements are all identical, we can probably do this in less lines